src/components/themed/EdgeOutlinedField.js

```diff
@@ -0,0 +1,428 @@
+  }
+
+  const handlePlaceholderLayout = useCallback(
+    ({ nativeEvent }) => {
+      const { width } = nativeEvent.layout
+      placeholderSize.value = width
+    },
+    [placeholderSize]
+  )
+
+  const handleContainerLayout = ({ nativeEvent }) => {
+    const { width, height } = nativeEvent.layout
+    containerWidth.value = width
+    setContainerHeight(height)
+  }
+
+  // handle value update
+  useEffect(() => {
+    if (_providedValue.length) placeholderMap.value = withTiming(1) // focused;
+    setValue(_providedValue)
+  }, [_providedValue, placeholderMap])
+  // error handling
+  useEffect(() => {
+    if (errorState()) {
+      colorMap.value = 2 // error -- no animation here, snap to color immediately
+    } else {
+      colorMap.value = isFocused() ? 1 : 0 // to active or inactive color if focused
+    }
+  }, [error, colorMap, errorState])
+
+  const animatedPlaceholderStyles = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateY: interpolate(placeholderMap.value, [0, 1], [0, -(paddingVertical + fontSize * placeholderScale)])
+      },
+      {
+        scale: interpolate(placeholderMap.value, [0, 1], [1, placeholderScale])
+      },
+      {
+        translateX: interpolate(placeholderMap.value, [0, 1], [0, -placeholderSize.value * placeholderSizeScale - hintLeftMargin])
+      }
+    ]
+  }))
+
+  const animatedPlaceholderTextStyles = useAnimatedStyle(() => ({
+    color: interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor])
+  }))
+
+  const animatedPlaceholderSpacerStyles = useAnimatedStyle(() => ({
+    width: interpolate(
+      placeholderMap.value,
+      [0, 1],
+      [containerWidth.value - placeholderSpacerWidthAdjust, containerWidth.value - placeholderSize.value * placeholderScale - placeholderSpacerAdjust],
+      Extrapolate.CLAMP
+    ),
+    backgroundColor: placeholderSize.value > 0 ? interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor]) : inactiveColor
+  }))
+  const cornerHeight = { height: containerHeight }
+
+  const animatedContainerStyle = useAnimatedStyle(() => {
+    const color = placeholderSize.value > 0 ? interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor]) : inactiveColor
+    return {
+      borderBottomColor: color,
+      borderLeftColor: color,
+      borderRightColor: color
+    }
+  })
+  const animatedLeftCornerStyle = useAnimatedStyle(() => {
+    const color = placeholderSize.value > 0 ? interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor]) : inactiveColor
+    return {
+      borderTopColor: color,
+      borderLeftColor: color
+    }
+  })
+  const animatedRightCornerStyle = useAnimatedStyle(() => {
+    const color = placeholderSize.value > 0 ? interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor]) : inactiveColor
+    return {
+      borderTopColor: color,
+      borderRightColor: color
+    }
+  })
```

Comment:
Avoid this code duplication.
Instead, create a function like this as a const:
```
const cornerStyle = (borders: string[]) => () => {
    const color = placeholderSize.value > 0 ? interpolateColor(colorMap.value, [0, 1, 2], [inactiveColor, activeColor, errorColor]) : inactiveColor
    return borders.reduce((res, border) => ({ ...res, [border]: color}), {})
  }
```

And then you can use it to create the animatedStyles like this:


```
const animatedLeftCornerStyle = useAnimatedStyle(cornerStyle(['borderTopColor', 'borderLeftColor']))
```